src/server/worldserver/CommandLine/CliRunnable.cpp

```diff
@@ -91,7 +93,9 @@ void utf8print(void* /*arg*/, char const* str)
     if (!Utf8toWStr(str, strlen(str), wtemp_buf, wtemp_len))
         return;
 
-    wprintf(L"%s", wtemp_buf);
+    char temp_buf[6000];
+    CharToOemBuffW(&wtemp_buf[0], &temp_buf[0], wtemp_len+1);
```

Comment:
could you post a test case we could use to test this ? did you investigate why wprintf doesn't work as expected in this case ?